src/docs/arch-connection.dox

```diff
@@ -0,0 +1,55 @@
+/*! @arch_page arch-connection Connection
+
+@section arch_conn_def Definition
+
+A connection is a handle to a WiredTiger database instance and only one connection can be opened at
+a time. Internally, a connection is represented by \c WT_CONNECTION.
+
+@section arch_conn_lifecycle Life cycle
+
+@subsection arch_conn_init Initialization
+
+A connection is initialized using WT_CONNECTION::wiredtiger_open. Given a set of configuration
+items (see @ref database_config), specific WiredTiger features can be enabled and their behavior
+tuned. Those features are for example related to @ref arch-eviction, @ref arch-logging, @ref
+arch-checkpoint, @ref arch-cache, statistics, etc. All the different available configuration
+settings are described in WT_CONNECTION::wiredtiger_open.
+
+WT_CONNECTION::wiredtiger_open also performs different sanity checks depending on the configuration
+item "create". When "create" is specified and a database does not exist already, a new database is
+created as well as specific WiredTiger files such as the turtle file and other metadata files. If a
+database already exists, WiredTiger will try to open it and check for the existence of the different
+required WiredTiger files. If "create" is not specified, WiredTiger expects a database where it is
+executed. In case of a failure during WT_CONNECTION::wiredtiger_open, the connection is not created
+and the returned error can be either \c WT_RUN_RECOVERY or \c WT_TRY_SALVAGE (see @ref
+error_handling) which means recovery might be required (see @ref command_line to learn more about
+it).
+
+Once the database has been successfully opened, worker threads are started to provide global
+services used during runtime. Those services consist of different threads to handle statistics,
+logging, eviction, checkpoint and cache management. The sweeping server that manages the active and
+inactive dhandles is started too, see @ref arch-dhandle for more information.
+
+Finally, before the connection is completely initialized, the database is set to a consistent state
+(see @ref arch-rts to read more about it).
+
+@subsection arch_conn_runtime Runtime
+
+During runtime, a number of actions can be executed using the connection. For instance, it is
```

Comment:
Something like : connection interface can be used to execute different database-wide runtime operations.